dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2119,65 @@ def _update_chi2(self):
 
         return chi2_mat
 
+    def _calc_nan_replication_dependent_properties(self, clean_samples):
+        """
+        Calculate metrics needed for replicating NaN values in capitalone/synthetic-data.
+
+        Metrics include class priors and class specific means of each column.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        """
+        if not self.options.synthetic_data.replicate_nan.is_enabled:
+            return
+
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        for col_id in range(len(self._profile)):
+            null_count = getattr(self._profile[col_id], "null_count")
+            if null_count == 0:
+                # No missing values to replicate
+                continue
+
+            # Partition data based on whether target column value is NaN or not
+            col_is_nan = data.iloc[:, col_id].isnull()
+            partition_nan = data[col_is_nan]
+            partition_not_nan = data[~col_is_nan]
+
+            # Drop the target column from partitions
+            partition_nan = partition_nan.drop(columns=col_id)
+            partition_not_nan = partition_not_nan.drop(columns=col_id)
+
+            # Calculate mean vectors of each partition
+            # i.e mean vectors of rows where target col is or is not NaN
+            mean_nan, mean_not_nan = (
+                partition_nan.mean().to_numpy(),
+                partition_not_nan.mean().to_numpy(),
+            )
```

Comment:
do we not have this info already from the col profiles? Is this only relative to the column we are iterating through?

Reply:
No we do not have this info pre calculated and this is pretty much the only extra calculation needed for replicating missing values in synthetic data. Basically we split the dataset into two partitions, one where the column we are processing (call it the `target column`) is NaN and where it is not NaN. Then take the means of each of the partitions (while not keeping the `target column` in the mean vectors calculated)